### DISCUSSION TITLE
Proper way to derive a control?

### DISCUSSION THREAD

**Discussion by himpactz** [AvaloniaUI/Avalonia#8801] [Q&A] [answered]
Hi, I am trying to derive from an existing control like Button or TextBlock but am not sure the "correct" way of doing so. 

1) Create a new .cs class and derive the Button class.  This approach seems to not work with IntelliSense as the class and its properties are not showing up in the axaml editor. However, it seems to work just fine.

2) Create a user control and then change the .cs file to deriving from Button. In this approach, the IntelliSense is working correctly. In this case, I have a axaml file that is not in used?

What is the right approach to deriving the class and have it work with IntelliSense?

Thanks,
Hai

**Comment by kekekeks** [comment, marked as answer]
The first approach is expected to work. Try rebuilding the project, our intellisense looks for types in the compiled assembly.

**Comment by himpactz** [comment]
Ah.. This is my bad... the 1st approach automatically set the class to internal and not public, thus why it didn't work. It is working now that I changed it to public.

Thanks!!!

**Reply by timunie** [reply]
Yeah Visual Studio now uses internal for new classes instead of public. In the past they used public. I really don't know why.

**Reply by rabbitism** [reply]
Their logic is like, public by default for class library and internal for application.

**Reply by timunie** [reply]
Thank you for clarification
